src/internal/digitalAsset.ts

```diff
@@ -181,3 +196,131 @@ export async function getTokenActivity(args: {
 
   return data.token_activities_v2;
 }
+
+export interface CreateCollectionOptions {
+  maxSupply?: AnyNumber;
+  mutableDescription?: boolean;
+  mutableRoyalty?: boolean;
+  mutableURI?: boolean;
+  mutableTokenDescription?: boolean;
+  mutableTokenName?: boolean;
+  mutableTokenProperties?: boolean;
+  mutableTokenURI?: boolean;
+  tokensBurnableByCreator?: boolean;
+  tokensFreezableByCreator?: boolean;
+  royaltyNumerator?: number;
+  royaltyDenominator?: number;
+}
+
+export async function createCollectionTransaction(
+  args: {
+    aptosConfig: AptosConfig;
+    creator: Account;
+    description: string;
+    name: string;
+    uri: string;
+    options?: GenerateTransactionOptions;
+  } & CreateCollectionOptions,
+): Promise<SingleSignerTransaction> {
+  const { aptosConfig, options, creator } = args;
+  const transaction = await generateTransaction({
+    aptosConfig,
+    sender: creator.accountAddress.toString(),
+    data: {
+      function: "0x4::aptos_token::create_collection",
+      arguments: [
+        // Do not change the order
+        new MoveString(args.description),
+        new U64(args.maxSupply ?? MAX_U64_BIG_INT),
+        new MoveString(args.name),
+        new MoveString(args.uri),
+        new Bool(args.mutableDescription ?? true),
+        new Bool(args.mutableRoyalty ?? true),
+        new Bool(args.mutableURI ?? true),
+        new Bool(args.mutableTokenDescription ?? true),
+        new Bool(args.mutableTokenName ?? true),
+        new Bool(args.mutableTokenProperties ?? true),
+        new Bool(args.mutableTokenURI ?? true),
+        new Bool(args.tokensBurnableByCreator ?? true),
+        new Bool(args.tokensFreezableByCreator ?? true),
+        new U64(args.royaltyNumerator ?? 0),
+        new U64(args.royaltyDenominator ?? 1),
```

Comment:
that looks pretty damn good 
cc @xbtmatt

Reply:
:D I know right! I know some people like the remote ABI but I love seeing the types